frontend/src/domains/routie/components/RoutieSection/RoutieSection.tsx

```diff
@@ -0,0 +1,61 @@
+import { useEffect } from 'react';
+
+import Flex from '@/@common/components/Flex/Flex';
+import Pill from '@/@common/components/Pill/Pill';
+import Text from '@/@common/components/Text/Text';
+import theme from '@/styles/theme';
+
+import { MOVING_EN_TO_KR } from '../../constants/translate';
+import { useRoutieContext } from '../../contexts/useRoutieContext';
+import { useRoutieValidateContext } from '../../contexts/useRoutieValidateContext';
+import { useCardDrag } from '../../hooks/useCardDrag';
+import { convertMetersToKilometers } from '../../utils/format';
+import RoutiePlaceCard from '../RoutiePlaceCard/RoutiePlaceCard';
+
+const RoutieSection = () => {
+  const { routiePlaces, routes, handleChangeRoutie } = useRoutieContext();
+  const getDragProps = useCardDrag(routiePlaces, handleChangeRoutie);
+  const { validateRoutie } = useRoutieValidateContext();
+
+  useEffect(() => {
+    if (!routiePlaces) return;
+
+    const isSequenceChanged = routiePlaces.some(
+      (item, index) => item.sequence !== index + 1,
+    );
+
+    const updateRoutiePlaces = async () => {
+      if (routiePlaces.length > 0) {
+        await validateRoutie();
+      }
+    };
+
+    if (isSequenceChanged) {
+      updateRoutiePlaces();
+    }
+  }, [routiePlaces]);
```

Comment:
`useEffect` 내부에 조건문이 많아 흐름 파악이 어려운 것 같아서, 보호구문을 미리 써줘도 좋을 것 같습니다

```suggestion
  useEffect(() => {
    if (!routiePlaces) return;
    if (routiePlaces.length <= 0) return;
    
    const isSequenceChanged = routiePlaces.some(
      (item, index) => item.sequence !== index + 1,
    );
    
    if (!isSequenceChanged) return;
    
    const updateRoutiePlaces = async () => {
        await validateRoutie();
    };

    updateRoutiePlaces();
   
  }, [routiePlaces]);
```

Reply:
말씀하신 보호구문을 위로 올려서 조건에 따라 얼리 리턴하는 방향으로 수정했습니다.